src/com/projectkorra/projectkorra/firebending/FireBlast.java

```diff
@@ -161,13 +162,10 @@ public boolean checkLocation(Block block) {
 				final BlastFurnace blastF = (BlastFurnace) block.getState();
 				blastF.setBurnTime((short) 800);
 				blastF.update();
-			} else if (block instanceof Campfire) {
-				final Campfire campfire = (Campfire) block.getBlockData();
-				if(!campfire.isLit()) {
-					if(block.getType() != Material.SOUL_CAMPFIRE || bPlayer.canUseSubElement(SubElement.BLUE_FIRE)) {
-						campfire.setLit(true);
-					}
-				}
+			} else if (block.getBlockData() instanceof Lightable) {
+				Lightable lightable = (Lightable) block.getBlockData();
+				lightable.setLit(true);
+				block.setBlockData(lightable);

```

Comment:
This new simpler implementation removes the previous behavior of requiring the blue fire subelement to light soul campfires. Is this a detail we care about maintaining?